test/YieldDirectorV2.test.js

```diff
@@ -0,0 +1,817 @@
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        await triggerRebase();
+
+        const donatedAmount = await gOhm.balanceTo("10000000");
+        await expect(await tyche.redeemableBalance("0")).is.equal(donatedAmount.add("1"));
+
+        const donationInfo = await tyche.depositInfo("0");
+        await expect(donationInfo.agnosticAmount).is.equal(principal);
+
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo.principalAmount);
+        await tyche.withdrawPrincipal("0", withdrawableBalance);
+
+        // Redeemable amount should be unchanged
+        await expect(await tyche.redeemableBalance("0")).is.equal(donatedAmount.add("1"));
+
+        // Trigger a few rebases
+        await triggerRebase();
+        await triggerRebase();
+        await triggerRebase();
+        await expect(await tyche.redeemableBalance("0")).is.equal(donatedAmount.add("1"));
+
+        await tyche.connect(bob).redeemAllYield();
+
+        await expect(await tyche.redeemableBalance(bob.address)).is.equal("0");
+    });
+
+    it("withdawable balance plus redeemable balance should equal deposited gOHM", async () => {
+        // Deposit 1 gOHM into Tyche and donate to Bob
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        await triggerRebase();
+
+        const donationInfo = await tyche.depositInfo("0");
+
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo.principalAmount);
+        const redeemable = await tyche.redeemableBalance("0");
+        await expect(withdrawableBalance.add(redeemable).toString()).is.equal(principal);
+    });
+
+    it("should withdraw tokens after recipient redeems", async () => {
+        // Deposit 1 gOHM into Tyche and donate to Bob
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        const donationInfo = await tyche.depositInfo("0");
+        await expect(donationInfo.agnosticAmount).is.equal(principal);
+
+        await triggerRebase();
+
+        const donationInfo1 = await tyche.depositInfo("0");
+        await expect(donationInfo1.agnosticAmount).is.equal(principal);
+
+        const redeemablePerRebase = await tyche.redeemableBalance("0");
+
+        await tyche.connect(bob).redeemYield("0");
+
+        await expect(await gOhm.balanceOf(bob.address)).is.equal(redeemablePerRebase);
+
+        const donationInfo2 = await tyche.depositInfo("0");
+        await expect(donationInfo2.agnosticAmount).is.equal("999000999000999000"); // 9.990~
+
+        await expect(await tyche.redeemableBalance("0")).is.equal("0");
+
+        // Second rebase
+        await triggerRebase();
+
+        await expect(await tyche.redeemableBalance("0")).is.equal(await gOhm.balanceTo("10000000"));
+
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo.principalAmount);
+        await tyche.withdrawPrincipal("0", withdrawableBalance);
```

Comment:
for one of these tests instead of putting withdrawableBalance here want to try put a super large number to see if it handles it by withdrawing everything?